DOCUMENTATION.md

```diff
@@ -3090,317 +2814,689 @@ Client.newHazelcastClient().then(function (client) {
+var winstonAdapter = {
+    logger: new (winston.Logger)({
+        transports: [
+            new (winston.transports.Console)()
+        ]
+    }),
 
-Client.newHazelcastClient().then(function (client) {
-    hazelcastClient = client;
-    return hazelcastClient.getMap('customers')
-}).then(function (mp) {
-    mapCustomers = mp;
-    return hazelcastClient.getMap('orders');
-}).then(function (mp) {
-    mapOrders = mp;
+    levels: [
+        'error',
+        'warn',
+        'info',
+        'debug',
+        'silly'
+    ],
 
-    // create the customer entry with customer id = 1
-    return mapCustomers.put(1, customer);
-}).then(function () {
-    // now create the orders for this customer
-    return mapOrders.putAll([
-        [new OrderKey(21, 1), order],
-        [new OrderKey(22, 1), order],
-        [new OrderKey(23, 1), order]
-    ]);
-});
-```
+    log: function (level, objectName, message, furtherInfo) {
+        this.logger.log(this.levels[level], objectName + ': ' + message, furtherInfo);
+    },
 
-For more details, see the [PartitionAware section](https://docs.hazelcast.org/docs/latest/manual/html-single/#partitionaware) in the Hazelcast IMDG Reference Manual.
+    error: function (objectName, message, furtherInfo) {
+        this.log(LogLevel.ERROR, objectName, message, furtherInfo);
+    },
 
-### 7.8.2. Near Cache
+    warn: function (objectName, message, furtherInfo) {
+        this.log(LogLevel.WARN, objectName, message, furtherInfo);
+    },
 
-Map entries in Hazelcast are partitioned across the cluster members. Hazelcast clients do not have local data at all. Suppose you read the key `k` a number of times from a Hazelcast client and `k` is owned by a member in your cluster. Then each `map.get(k)` will be a remote operation, which creates a lot of network trips. If you have a map that is mostly read, then you should consider creating a local Near Cache, so that reads are sped up and less network traffic is created.
+    info: function (objectName, message, furtherInfo) {
+        this.log(LogLevel.INFO, objectName, message, furtherInfo);
+    },
 
-These benefits do not come for free, please consider the following trade-offs:
+    debug: function (objectName, message, furtherInfo) {
+        this.log(LogLevel.DEBUG, objectName, message, furtherInfo);
+    },
 
-- If invalidation is enabled and entries are updated frequently, then invalidations will be costly.
+    trace: function (objectName, message, furtherInfo) {
+        this.log(LogLevel.TRACE, objectName, message, furtherInfo);
+    }
 
-- Near Cache breaks the strong consistency guarantees; you might be reading stale data.
+};
+cfg.customLogger = winstonAdapter;
+```
 
-- Clients with a Near Cache will have to hold the extra cached data, which increases memory consumption.
+Note that it is not possible to configure custom logging via declarative configuration.
 
-Near Cache is highly recommended for maps that are mostly read.
+## 7.10. Defining Client Labels
```

Comment:
Does it make sense to change the section name to `Defining Instance Name and Client Labels` and describe setting client instance name here as well?

Reply:
I think the labels and instance name are different concepts so I added a new section for that(7.11)

Reply:
A new section will work as well. Thanks for adding it!